x-pack/plugins/cloud_security_posture/public/pages/benchmarks/benchmarks.tsx

```diff
@@ -4,49 +4,135 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import { EuiPageHeaderProps, EuiButton, EuiSpacer } from '@elastic/eui';
-import React from 'react';
+import {
+  EuiFieldSearch,
+  EuiPageHeaderProps,
+  EuiButton,
+  EuiSpacer,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTextColor,
+  EuiText,
+} from '@elastic/eui';
+import React, { useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
 import { allNavigationItems } from '../../common/navigation/constants';
 import { useCspBreadcrumbs } from '../../common/navigation/use_csp_breadcrumbs';
-import { CspLoadingState } from '../../components/csp_loading_state';
 import { CspPageTemplate } from '../../components/page_template';
 import { BenchmarksTable } from './benchmarks_table';
-import { ADD_A_CIS_INTEGRATION, BENCHMARK_INTEGRATIONS, LOADING_BENCHMARKS } from './translations';
+import { ADD_A_CIS_INTEGRATION, BENCHMARK_INTEGRATIONS } from './translations';
 import { useCspBenchmarkIntegrations } from './use_csp_benchmark_integrations';
+import { pagePathGetters } from '../../../../fleet/public';
+import { useKibana } from '../../common/hooks/use_kibana';
+import { extractErrorMessage } from '../../../common/utils/helpers';
+import { SEARCH_PLACEHOLDER } from './translations';
+
+const integrationPath = pagePathGetters.integrations_all({ searchTerm: 'CIS' }).join('');
+
+const AddCisIntegrationButton = () => {
+  const { http } = useKibana().services;
+
+  return (
+    <EuiButton fill iconType="plusInCircle" href={http.basePath.prepend(integrationPath)}>
+      {ADD_A_CIS_INTEGRATION}
+    </EuiButton>
+  );
+};
+
+const BenchmarkEmptyState = ({ name }: { name: string }) => (
+  <div>
+    <EuiSpacer size="l" />
+    {name && (
+      <EuiText>
+        <strong>
+          <FormattedMessage
+            id="xpack.csp.benchmarks.integrationsNotFoundMessage"
+            defaultMessage='No benchmark integrations found for "{name}"'
+            values={{ name }}
+          />
+        </strong>
+      </EuiText>
+    )}
+    <EuiSpacer size="s" />
+    <EuiText>
+      <EuiTextColor color="subdued">
+        <FormattedMessage
+          id="xpack.csp.benchmarks.integrationsNotFoundWithFiltersMessage"
+          defaultMessage="We weren’t able to find any benchmark integrations with the above filters."
```

Comment:
@einatjacoby 

from what i gather, if the user didn't install an integration, they would get the [blank page](https://github.com/elastic/kibana/pull/126750)

if they did install at least 1, and then run a search by name, we should show something like the following in case we have no results: "No integrations found for 'query'"

so IMO - there is no need to split to with/without filter, as we won't get to the without case anyway. 

but given that, we should probably modify the current messages.